pkg/reconciler/intevents/pullsubscription/reconciler.go

```diff
@@ -338,13 +337,13 @@ func (r *Base) reconcileDataPlaneResources(ctx context.Context, ps *v1.PullSubsc
 }
 
 func (r *Base) GetOrCreateReceiveAdapter(ctx context.Context, desired *appsv1.Deployment, ps *v1.PullSubscription) (*appsv1.Deployment, error) {
-	existing, err := r.getReceiveAdapter(ctx, ps)
-	if err != nil && !apierrors.IsNotFound(err) {
-		logging.FromContext(ctx).Desugar().Error("Unable to get an existing Receive Adapter", zap.Error(err))
-		ps.Status.MarkDeployedUnknown("ReceiveAdapterGetFailed", "Error getting the Receive Adapter: %s", err.Error())
-		return nil, err
-	}
-	if existing == nil {
+	existing, err := r.KubeClientSet.AppsV1().Deployments(ps.Namespace).Get(ctx, desired.Name, metav1.GetOptions{})
+	if err != nil {
```

Comment:
Changed to 
```
if err == nil {
   return existing, nil
}
// logic without indent
```

Reply:
Thanks for the comment! Does this change cause any differences for the logic?
It seems like if changing to the way you suggested, it will `return existing, nil`  again at the end after creating the new deployment, I'd rather keep it as this.

Reply:
It won't change the logic, this is a common way to remove unnecessary nested logic. Eg. the if body might later become big with more nested logic, and this is one way of keeping the code cleaner.

Reply:
Makes sense! Thanks for the explanation :)